### PR TITLE
More work to make tools-common librarified

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -62,82 +60,104 @@ module.exports.depcheck = function depcheck(options) {
...
+  task(gulp, 'test', ['build'], () =>
     gulp.src('test/**/*_test.js', {read: false})
         .pipe(mocha({
           ui: 'tdd',
           reporter: 'spec',
         }))
   );
 }
+
+if (__dirname === process.cwd()) {
```
Comment (justinfagnani) [inline gulpfile.js]: what does this do?

File: gulpfile.js [hunk 2]
```diff
@@ -10,37 +10,35 @@
...
+module.exports.depcheck = function depcheck(gulp, options) {
+  const depcheck_lib = require('depcheck');
   const defaultOptions = {stickyDeps: new Set()};
   options = Object.assign({}, defaultOptions, options);
 
-  task('depcheck', () => {
+  task(gulp, 'depcheck', () => {
     return new Promise((resolve, reject) => {
-      depcheck_lib(__dirname, {ignoreDirs: []}, resolve);
+      depcheck_lib(process.cwd(), {ignoreDirs: []}, resolve);
```
Comment (justinfagnani) [inline gulpfile.js]: I don't think cwd is the right thing here... we really want to __dirname of the calling gulpfile

Comment (rictic) [inline gulpfile.js]: In a very sensible move, the gulp authors have ensured that they are one and the same. If you run gulp in a subdirectory the cwd will be moved to the directory of the gulpfile.

I didn't find much documentation of this, but here's one place: https://github.com/gulpjs/gulp/issues/523

It also logs this fact to stdout when it does change the cwd.

File: gulpfile.js (hunk 1 above)
Comment (rictic) [inline gulpfile.js]: Added a comment.

**Comment by rictic** [comment]
(btw, I'm still not a collaborator in this repo)

File: gulpfile.js (hunk 2 above)
Comment (justinfagnani) [inline gulpfile.js]: Wow, that's awesome to know, and unawesome to only document it in a closed issue. Can you comment?

**Comment by justinfagnani** [comment]
you should be good to go on repo access now

File: gulpfile.js (hunk 2 above)
Comment (rictic) [inline gulpfile.js]: Added a comment

**Comment by justinfagnani** [comment]
Can you explain this change a little bit? In what cases do we need to pass `gulp` as a parameter, vs directly require gulp. Is there a comon situation where the importing gulpfile and this would import different versions? Could this be solved by adding gulp as a peer dependency?

**Comment by rictic** [comment]
Cool, peer dependency works! (just not with npm link)

File: gulp-tasks.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+const fs = require('fs-extra');
```
Comment (justinfagnani) [inline gulp-tasks.js:13]: sort imports by package name, standardize on single quotes

File: gulp-tasks.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+
+module.exports.buildAll = function(options) {
+  module.exports.clean(options);
+  module.exports.init(options);
+  module.exports.lint(options);
+  module.exports.build(options);
+
+  task('build-all', (done) => {
+    runSeqBase.use(gulp)('clean', 'init', 'lint', 'build', done);
```
Comment (justinfagnani) [inline gulp-tasks.js]: do you need the `use()` call now?

**Comment by justinfagnani** [comment]
Can we add a gulpfile that we can copy into a project with all the tasks?

File: gulp-tasks.js (hunk 3 above)
Comment (rictic) [inline gulp-tasks.js:13]: Done

File: gulp-tasks.js (hunk 4 above)
Comment (rictic) [inline gulp-tasks.js]: Nope! Removed

**Comment by rictic** [comment]
See gulpfile.js for the complete gulpfile that we can copy elsewhere.

**Comment by justinfagnani** [comment]
LGTM

**Pull request merged**